lib/serial_terminal.pm

```diff
@@ -84,12 +84,12 @@ sub login {
     # newline nudges the guest to display the login prompt, if this behaviour
     # changes then remove it
     type_string("\n");
-    wait_serial(qr/login:\s*$/i);
+    die 'Failed to wait for login prompt' unless wait_serial(qr/login:\s*$/i);
     type_string("$user\n");
-    wait_serial(qr/Password:\s*$/i);
+    die 'Failed to wait for password prompt' unless wait_serial(qr/Password:\s*$/i);
     type_password;
     type_string("\n");
-    wait_serial(qr/$escseq* \w+:~\s\# $escseq* \s*$/x);
+    die 'Failed to confirm that login successful' unless wait_serial(qr/$escseq* \w+:~\s\# $escseq* \s*$/x);
```

Comment:
nit: 'Failed to confirm that login *was* successful'

Reply:
Done